src/main/java/io/kestra/storage/s3/S3Storage.java

```diff
@@ -231,15 +231,21 @@ public URI createDirectory(String tenantId, URI uri) throws IOException {
 
     private void mkdirs(String path) throws IOException {
         try {
-            if (!path.endsWith("/") && path.lastIndexOf('/') > 0) {
-                path = path.substring(0, path.lastIndexOf('/') + 1);
+            if(!path.startsWith("/")) {
+                path = "/" + path;
```

Comment:
path are always build by the `getPath` method, we may make sure that it starts with a / there to avoid checking it everywhere.

Reply:
Maybe we want to be able to list on "" to eventually get the "/" directory ? That's the reason why I isolated it for dir creation